### PR TITLE
fixes GH workflows for external PRs

### PR DESCRIPTION
**Pull request by zzantozz**
The `pull_request` trigger on workflows has limited permissions, since
it potentially runs external, untrusted code. Therefore no secrets are
exposed to it, like the token for adding a comment to the PR. One way
around this is to trigger a second, trusted workflow from the PR
workflow to do the work.

See
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
for a full description.

### CODE REVIEW

File: .github/workflows/on-pr.yml [hunk 1]
```diff
@@ -26,29 +26,29 @@ jobs:
...
         continue-on-error: true
 
-      - uses: actions/github-script@0.9.0
-        env:
-          PLAN: "${{ steps.plan.outputs.stdout }}"
+      # See PR details usage in pr-tagback.yml
+      - name: store PR details as artifact
+        run: |
+          mkdir pr-details
+          echo ${{ github.event.number }} > pr-details/number.txt
```
Comment (iWebi) [inline .github/workflows/on-pr.yml:33]: what happens when multiple PRs are raised simultaneously? Although its rare for multiple PRs against this main repo.

**Review by iWebi — DISMISSED** [review]

File: .github/workflows/on-pr.yml (hunk 1 above)
Comment (iWebi) [reply .github/workflows/on-pr.yml:33]: never mind. These temporary files are part of the container running the workflow and should be unique to each PR, I believe.

Comment (zzantozz) [reply .github/workflows/on-pr.yml:33]: Oh... This won't work.Yes, artifacts are specific to a workflow. They can only be shared between jobs in a single workflow. This PR uses two different workflows. I can't just do a simple "download artifact" in the second workflow.

Comment (zzantozz) [reply .github/workflows/on-pr.yml:33]: The [new commit](https://github.com/rax-maas/rax-maas-github-settings/pull/3/commits/bea8d6407e43bb223ff2d9e055a1cfc98f29a49f) should fix this. The script was in that GH security blog post I mentioned, but I didn't realize the need for it before.

**Review by iWebi — APPROVED** [review]
👍

**Pull request merged**